lib/src/Pos/Diffusion/Full/Block.hs

```diff
@@ -454,8 +456,10 @@ handleGetBlocks logTrace logic recoveryHeadersMessage oq = listenerConv logTrace
                      " blocks to "%build%" one-by-one")
                     (length hashes) nodeId )
                 for_ hashes $ \hHash ->
-                    getBlock logic hHash >>= \case
-                        Just b -> send conv $ MsgBlock b
+                    getSerializedBlock logic hHash >>= \case
+                        -- TODO: we should get lazy bytestring from the db layer to
+                        -- stream the bytes to the network
```

Comment:
Yes indeed we should. Would that be a huge change? Maybe we should make a ticket for this.

Reply:
I am not sure, maybe not, let me check.

Reply:
On second thought, I think it's fine that rocksdb gives and takes strict bytestrings (it has to take them, since the write needs one pointer). Here we can just use lazy bytestrings to construct the `MsgSerializedBlock` encoding, and avoid one copy.

Reply:
For `RocksDB` it's true, but for `MonadDB` which writes to disc it may be fine to take lazy `ByteString`.  The only thing is that it may limit future options what to use for the block/undo storage layer.

Reply:
Yeah I think as a rule it's better to take lazy bytestrings, since `fromStrict` is cheap but `toStrict` is not.

Don't worry about limiting future options, we can always change `MonadDB`. Nothing is sacred.

Reply:
Ok, so I'll update the `MonadDB`  to use lazy `ByteString`, anyway future option should be lazy ;)